bundles/config/org.eclipse.smarthome.config.discovery.usbserial/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/internal/UsbSerialDiscoveryService.java

```diff
@@ -0,0 +1,179 @@
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link DiscoveryService} for discovering USB devices with an associated serial port.
+ * <p/>
+ * This discovery service is intended to be used by bindings that support USB devices, but do not directly talk to the
+ * USB devices but rather use a serial port for the communication, where the serial port is provided by an operating
+ * system driver outside the scope of Eclipse SmartHome. Examples for such USB devices are USB dongles that provide
+ * access to wireless networks, like, e.g., Zigbeee or Zwave dongles.
+ * <p/>
+ * This discovery service provides functionality for discovering added and removed USB devices and the corresponding
+ * serial ports. The actual {@link DiscoveryResult}s are then provided by {@link UsbSerialDiscoveryParticipant}s, which
+ * are called by this discovery service whenever new devices are detected or devices are removed. Such
+ * {@link UsbSerialDiscoveryParticipant}s should be provided by bindings accessing USB devices via a serial port.
+ * <p/>
+ * This discovery service requires a component implementing the interface {@link UsbSerialDiscovery}, which performs the
+ * actual serial port and USB device discovery (as this discovery might differ depending on the operating system).
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@NonNullByDefault
+@Component(immediate = true, service = { DiscoveryService.class,
+        UsbSerialDiscoveryService.class }, configurationPid = "discovery.usbserial")
+public class UsbSerialDiscoveryService extends AbstractDiscoveryService implements UsbSerialDiscoveryListener {
+
+    private final Logger logger = LoggerFactory.getLogger(UsbSerialDiscoveryService.class);
+
+    private final Set<UsbSerialDiscoveryParticipant> discoveryParticipants = new CopyOnWriteArraySet<>();
+
+    private final Set<UsbSerialDeviceInformation> previouslyDiscovered = new CopyOnWriteArraySet<>();
+
+    @NonNullByDefault({})
+    private UsbSerialDiscovery usbSerialDiscovery;
+
+    public UsbSerialDiscoveryService() {
+        super(5);
+    }
+
+    @Override
+    @Activate
+    protected void activate(@Nullable Map<String, @Nullable Object> configProperties) {
+        super.activate(configProperties);
+        usbSerialDiscovery.registerDiscoveryListener(this);
+        if (isBackgroundDiscoveryEnabled()) {
+            usbSerialDiscovery.startBackgroundScanning();
+        }
+    }
+
```

Comment:
This is missing a 

```java

@Override
@Deactivate
protected void deactivate() {
  super.deactivate();
}
```

Reply:
I will add this (as discussed, to make explicit that the super method is called, although it would also be called without this overridden method).